### PR TITLE
Remove host, protocol options, only support url in RequestFactory

### PR DESCRIPTION
**Pull request by LionOps**

### CODE REVIEW

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by kahnvex** [comment]
I'm confused, why relative path?

File: test/pixabay-spec.js [hunk 1]
```diff
@@ -21,11 +21,12 @@ var username = 'username';
 var key = 'key';
 
 var pixabayUrl = 'https://pixabay.com/api';
+var relativeURL = '/pixabay/api';
```
Comment (kahnvex) [inline test/pixabay-spec.js]: That is an absolute path.

**Comment by LionOps** [comment]
A better name is welcomed.

**Comment by kahnvex** [comment]
A relative path is something like `./relative/path`, `relative/path`, or `../some/path`. Anything starting with `/` is _not_ a relative path. I think we should simply store one field, that we parse using the URL module, when we add a pefix, querystring, etc.

That field can have the protocol and host, or leave it off.

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: src/retriever.js [hunk 2]
```diff
@@ -7,9 +7,8 @@ var url = require('url');
 
 function Retriever(options) {
   options = options || {};
-  this._host = options.host || 'pixabay.com';
-  this._path = options.path || 'api';
-  this._protocol = options.protocol || 'https';
+
+  this._path = options.url || 'https://pixabay.com/api';
```
Comment (kahnvex) [inline src/retriever.js]: I know I'm really harping on naming, but can we just call this a URL all the way through?

Comment (LionOps) [inline src/retriever.js]: If you give me a name for the function `_url()`, sure :)

Comment (euwest) [inline src/retriever.js]: `_urlPromise()`

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by Hkly** [comment]
:+1:

**Pull request merged**